test/ProblemDetailsResponseFactoryTest.php

```diff
@@ -176,4 +178,14 @@ public function testFactoryRendersPreviousExceptionsInDebugMode() : void
         $this->assertEquals(101010, $payload['exception']['stack'][0]['code']);
         $this->assertEquals('first', $payload['exception']['stack'][0]['message']);
     }
+
+    public function testFragileDataInExceptionMessageShouldBeHiddenInResponseBodyInNoneDebugMode()
+    {
+        $fragileMessage = 'Your SQL or password here';
+        $exception = new \Exception($fragileMessage);
+
+        $response = $this->factory->createResponseFromThrowable($this->request->reveal(), $exception);
+
+        $this->assertNotContains($fragileMessage, (string) $response->getBody());
```

Comment:
Injecting the response is not enough, as we need to be able to populate the content, and the only way to do that _safely_ is to provide a new `StreamInterface` instance to a clone of the prototype.

In terms of JSON vs XML, we _could_ go the route I took in [weierophinney/hal](https://github.com/weierophinney/hal) and have a `Renderer` subcomponent. If we do that, however, then we need to be able to pass both of those (both JSON and XML renderers) to the response factory, too. It's a coin toss as to which would be "better"; you end up with the same number of arguments.